src/multi/mod.rs

```diff
@@ -337,7 +337,7 @@ where
 /// [`cut`][crate::combinator::cut].
 ///
 /// # Arguments
-/// * `sep` Parses the separator between list elements.
+/// * `sep` Parses the separator between list elements. Must be consuming. 
```

Comment:
`many0` has a similar note that is separate from the Arguments list

> Note: if the parser passed in accepts empty inputs (like alpha0 or digit0), many0 will return an error, to prevent going into an infinite loop

I'm assuming it'd be good to be consistent.

Reply:
Whether a parser consumes some input is clearly an important characteristic. As such, it would be good if that characteristic could be put into the type system.